File: src/main/java/org/example/commands/AddCommand.java
package org.example.commands;

import org.example.collection.MusicBand;
import org.example.collection.builder.BuilderMusicBand;
import org.example.exception.ArgumentCountError;
import org.example.exception.ScriptExecutionError;
import org.example.handlers.ICollectionController;
import org.example.handlers.RunHandler;

/**
 * Команда add. Позволяет добавить новый элемент в коллекцию.
 */
public class AddCommand extends ACommand {
    private ICollectionController collectionController;

    public AddCommand(ICollectionController collectionController) {
        super("add", "Команда позволяет добавить новый элемент.", 0);
        this.collectionController = collectionController;
    }

    /**
     * Метод для добавления элемента в коллекцию
     *
     * @param args аргументы
     * @throws ArgumentCountError   если количество аргументов не совпадает
     * @throws ScriptExecutionError если произошла ошибка во время выполнения скрипта
     */
    @Override
    public void execute(String[] args) throws ArgumentCountError, ScriptExecutionError {
        valideCountsArgument(args);
        MusicBand musicBand = new BuilderMusicBand().build();
        collectionController.add(musicBand);
        if (!RunHandler.mode()) {
            System.out.println("Элемент успешно добавлен");
        }
    }
}

File: server/src/main/kotlin/commands/AddCommand.kt
package commands

import ICollectionController
import collection.MusicBand
import java.util.concurrent.locks.ReentrantLock

class AddCommand(private val collectionController: ICollectionController<*>, private val lock: ReentrantLock) :
    ACommand("add", "Команда позволяет добавить новый элемент.", "Элемент успешно добавлен в коллекцию.") {
    override fun doIt(
        arg: Any?,
        id: Int,
    ) {
        try {
            lock.lock()
            collectionController.add(arg!! as MusicBand, id)
        } finally {
            lock.unlock()
        }
    }
}
